File: scratch/search_os.js
import fs from 'fs';
import path from 'path';

const filePath = 'c:/Users/Henrique - PC/Desktop/Projetos Dev/crm-mdr/src/pages/ServiceOrders.tsx';

const keywords = ['offerRedirectToSales'];

try {
  const content = fs.readFileSync(filePath, 'utf-8');
  const lines = content.split('\n');
  console.log(`Total lines: ${lines.length}`);

  const results = {};
  keywords.forEach(kw => {
    results[kw] = [];
  });

  lines.forEach((line, idx) => {
    const lineNum = idx + 1;
    keywords.forEach(kw => {
      const regex = new RegExp('\\b' + kw + '\\b', 'i');
      if (regex.test(line)) {
        results[kw].append ? results[kw].append({ lineNum, text: line.trim() }) : results[kw].push({ lineNum, text: line.trim() });
      }
    });
  });

  keywords.forEach(kw => {
    const matches = results[kw];
    if (matches && matches.length > 0) {
      console.log(`\n=== Matches for '${kw}' (first 10): ===`);
      matches.slice(0, 10).forEach(m => {
        console.log(`  Line ${m.lineNum}: ${m.text}`);
      });
    }
  });

} catch (err) {
  console.error('Error:', err);
}

File: scratch/search_os.py
import re

file_path = 'c:/Users/Henrique - PC/Desktop/Projetos Dev/crm-mdr/src/pages/ServiceOrders.tsx'

keywords = ['venda', 'pagamento', 'finalizar', 'concluir', 'status', 'faturar', 'faturamento', 'checkout', 'navigate']

with open(file_path, 'r', encoding='utf-8') as f:
    lines = f.readlines()

print(f"Total lines: {len(lines)}")

results = {}
for kw in keywords:
    results[kw] = []

for idx, line in enumerate(lines):
    for kw in keywords:
        if re.search(r'\b' + re.escape(kw) + r'\b', line, re.IGNORECASE):
            results[kw].append((idx + 1, line.strip()))

for kw, matches in results.items():
    if matches:
        print(f"\n=== Matches for '{kw}' (first 10): ===")
        for line_num, content in matches[:10]:
            print(f"  Line {line_num}: {content}")
